src/odemis/driver/test/simulated_test.py

```diff
@@ -289,5 +290,29 @@ def test_wrong_moveAbs(self):
         with self.assertRaises(ValueError):
             self.dev.moveAbs({"pressure": -5})
 
+
+class GenericComponentTest(unittest.TestCase):
+
+    def test_creation_complete(self):
+        simulated.GenericComponent(name="test_component", role="test", vas={"vaRange":
+                                                                                {"value": 0.1, "readonly": True,
+                                                                                 "unit": "", "range": [0, 1]},
+                                                                            "vaChoices":
+                                                                                {"value": 1,
+                                                                                 "choices": set(range(0, 10))},
+                                                                            "vaBool": {"value": True}},
+                                   axes={"x": {"range": (-0.2, 0.2), "unit": "m"},
+                                         "gripper": {"choices": {'open': False, 'closed': True}}})
+
+    def test_creation_vas_only(self):
+        simulated.GenericComponent(name="test_component", role="test", vas={"vaRange":
+                                                                                {"value": 0.1, "readonly": True,
+                                                                                 "unit": "", "range": [0, 1]},
+                                                                            "vaChoices":
+                                                                                {"value": 1,
+                                                                                 "choices": set(range(0, 10))},
+                                                                            "vaBool": {"value": True}})
+
+
```

Comment:
Please check that the VAs are indeed there, with the intended values and attributes. It should also be possible to set them (excepted for the readonly one).